edb/server/compiler/compiler.py

```diff
@@ -1822,17 +1836,133 @@ def _compile_ql_query(
 
     result_cardinality = enums.cardinality_from_ir_value(ir.cardinality)
 
+    # This low-hanging-fruit is temporary; persistent cache should cover all
+    # cacheable cases properly in future changes.
+    use_persistent_cache = (
+        ctx.persistent_cache and
+        cacheable and
+        not ctx.bootstrap_mode and
+        script_info is None
+    )
+
     sql_res = pg_compiler.compile_ir_to_sql_tree(
         ir,
         expected_cardinality_one=ctx.expected_cardinality_one,
         output_format=_convert_format(ctx.output_format),
         backend_runtime_params=ctx.backend_runtime_params,
         expand_inhviews=options.expand_inhviews,
+        detach_params=bool(use_persistent_cache and debug.flags.func_cache),
     )
 
-    sql_text = pg_codegen.generate_source(sql_res.ast)
+    sql_ast = sql_res.ast
+
+    pg_debug.dump_ast_and_query(sql_ast, ir)
```

Comment:
Remove this dump

Reply:
Please put this dump back; it's conditional in `pg_debug`

Reply:
Actually I just did it.
(I was doing some testing of something unrelated on top of this branch, to avoid needing to rebuild, and ran into this being gone)